feat(medic/cht-core/issues/9827): SSO setup documentation (#1875)
medic/cht-docs: content/en/hosting/SSO/_index.md

---
title: Single Sign-On
linkTitle: SSO
weight: 200
description: Setting up an single sign on with the CHT
---

CHT version 4.20.0 introduces support for Single Sign-On (SSO).  This enables deployments to authenticate users outside of the CHT.  While the feature was extensively tested with the free and open source [KeyCloak](https://www.keycloak.org/) and the commercial [Microsoft Entra](https://learn.microsoft.com/en-us/entra/fundamentals/what-is-entra) providers, it should work with any [OpenID Connect](https://openid.net/) (OIDC) compliant identity provider.

![sso-login-flow.svg](sso-login-flow.svg)

> 

Author: ah - gotcha - indeed it is only a few key steps that are needed.  Lemme have a go at adding them !

---

> title: Single Sign On

Reviewer: ```suggestion
title: Single Sign-On
```

linkTitle: SSO
weight: 200
---

> {{< hextra/hero-subtitle >}}

Author: thanks!  fixed in 35c266a5

  Setting up an single sign on with the CHT
{{< /hextra/hero-subtitle >}}

> CHT version 4.20.0 introduces support for single sign on (SSO).  This enables deployments to authenticate users outside of the CHT.  While the feature was extensively tested with the free and open source [KeyCloak](https://www.keycloak.org/) and the commercial [Microsoft Entra](https://learn.microsoft.com/en-us/entra/fundamentals/what-is-entra) providers, it should work with any [OpenID Connect](https://openid.net/) (OIDC) compliant identity provider.

Reviewer: ```suggestion
CHT version 4.20.0 introduces support for Single Sign-On (SSO).  This enables deployments to authenticate users outside of the CHT.  While the feature was extensively tested with the free and open source [KeyCloak](https://www.keycloak.org/) and the commercial [Microsoft Entra](https://learn.microsoft.com/en-us/entra/fundamentals/what-is-entra) providers, it should work with any [OpenID Connect](https://openid.net/) (OIDC) compliant identity provider.
```

---
title: Single Sign-On
linkTitle: SSO
weight: 200

> description: Setting up an single sign on with the CHT

Reviewer 2: ```suggestion
description: Setting up Single Sign On with the CHT
relatedContent: >
  building/concepts/access/#single-sign-on-oidc-login
  building/reference/api/#login-by-oidc
  building/reference/app-settings/oidc_provider
```

- [Microsoft Entra ID](https://learn.microsoft.com/en-us/entra/fundamentals/what-is-entra) - Paid, cloud-based identity and access management service.

![sso-login-flow.svg](sso-login-flow.svg)

## Quick Start

1. Create an OIDC Provider client with the redirect URL of `https://<CHT_URL>/medic/login/oidc`. Grab the secret for step 3 below. Be sure to replace `CHT_URL` with your real URL.
2. Add the Client App ID (`APP_ID`), OIDC URL (`WELL_KNOWN_OIDC_URL`) and CHT URL (`CHT_URL`) to the app settings and upload it with CHT Conf to your CHT instance. Be sure to replace `APP_ID` and `WELL_KNOWN_OIDC_URL` and `CHT_URL` with your real URLs:

    ```yaml
    "oidc_provider": {
      "client_id": "<APP_ID>",
      "discovery_url": "https://<WELL_KNOWN_OIDC_URL>"
    },
    "app_url": "https://<CHT_URL>/"
    ```
3. Upload the secret from step 1 to the CHT. Be sure to replace `CHT_URL` , `USER`,  `PASSWORD` and `SECRET` with the correct values: `curl -X PUT https://<USER>:<PASSWORD>@<CHT_URL>/api/v1/credentials/oidc:client-secret -H "Content-Type: text/plain" --data "<SECRET>"`

> 4. Ensure CHT users have their "SSO Email Address" match the OIDC provider's account 

Author: fixed in c9515543
